packages/lab/src/Flow/Node/BaseNode.tsx

```diff
@@ -231,21 +231,37 @@ export const HvFlowBaseNode = ({
           </div>
 
           <div className={classes.inputsContainer}>
-            {inputs?.map((input) => (
-              <div className={classes.inputContainer} key={input.id}>
-                <Handle
-                  type="target"
-                  isConnectableStart={false}
-                  id={input.id}
-                  position={Position.Left}
-                />
-                <HvTypography>{input.label}</HvTypography>
-                {input.isMandatory &&
-                  !isInputConnected(id, "target", input.id!, inputEdges) && (
-                    <div className={classes.mandatory} />
-                  )}
-              </div>
-            ))}
+            {inputs?.map((input, idx) => {
+              const handleId = input.id ?? idx.toString();
+              const isConnected = isInputConnected(
+                id,
+                "target",
+                handleId,
+                inputEdges
+              );
```

Comment:
We actually no longer need to calculate the `handleId` and we can always use `input.id` since we generate ids for the inputs and outputs that don't have one defined. Thus, we can do: (same for the outputs below)

```tsx
const isConnected = isInputConnected(id, "target", input.id!, inputEdges)
```